core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -1987,6 +1989,21 @@ protected final void check(
     assertTrue(filterCollation.satisfies(sortCollation));
   }
 
+  @Test public void testRelShuttleForLogicalTableModify() {
+    String sql = "insert into emp select * from emp";
+    LogicalTableModify rel = (LogicalTableModify) tester.convertSqlToRel(sql).rel;
+    List<RelNode> rels = new ArrayList<>();
+    final RelShuttleImpl visitor = new RelShuttleImpl() {
+      @Override public RelNode visit(LogicalTableModify modify) {
+        rels.add(modify);
+        return super.visit(modify);
```

Comment:
I am thinking the right test to the code should be:

```
@Override public RelNode visit(LogicalTableModify modify) {
      Rel visitedRel = super.visit(modify);        
      rels.add(visitedRel);
       return visitedRel;
}

```

Reply:
For this test they are equivalent, but i think you are right